cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java

```diff
@@ -70,7 +70,7 @@
    * @return A list of {@link SecureKeyListEntry} for all the keys visible to the user under the given namespace.
    * @throws NamespaceNotFoundException If the specified namespace does not exist.
    * @throws IOException If there was a problem reading from the store.
-   * @throws Exception If we fail to create a filter for the current principal.
+   * @throws UnauthorizedException If the user does not have READ permissions on the namespace.
```

Comment:
wait, this is wrong, right? This never throws UnauthorizedException. It only returns entities you have some privileges for.